### PR TITLE
Add bindings scope blog post

### PR DESCRIPTION
**Pull request by thunderbiscuit**
This is a draft of a blog post explaining the position that the BDK project intends to take with regards to developing bindings, and outlines how we intend on becoming leaders and helping other libraries in the Rust bitcoin ecosystem produce bindings of their own.

### CODE REVIEW

**Comment by thunderbiscuit** [comment]
One issue I have with the post as-is is that the repositories linked at the bottom are all under my control but the language makes it feel like it's the whole BDK project that intends to support them. We might need to clean this up a bit? Either bring some of them under the org, or rework the language a bit to make it clear that they're more my personal projects? The template and examples could easily be under BDK, bitcoin-frontier should not. Let me know what you think.

File: docs/_blog/bindings-scope.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+1. Features that are part of crates "upstream" of BDK (rust-bitcoin, rust-miniscript)
+2. Features that are not but that have Rust crates and would be useful on mobile (payjoin, coinjoin implementations, silent payments, BIP-47)
+
+## Current architecture
+The current architecture for the BDK bindings is more or less wrapping the bdk, rust-bitcoin, and rust-miniscript crates and exposing an API that allows users to leverage them similarly to how they would BDK in Rust if they were using it in a Rust project.
+
+While we started with a simplified version of the Rust BDK API, over time users asked for more and more functionality, and exposing some of the underlying rust-bitcoin constructs became important. This makes sense, and indeed users of the bitcoin development kit in Rust can access to all the related APIs by simply importing rust-bitcoin and rust-miniscript, hence our desire to accomodate these use cases as well. However, this is currently done all in one "bindings" library (i.e. if you import `bdk-android` in a project, you'll have access to an API that is mostly bdk-based, but also contains a bit of rust-bitcoin and rust-miniscript).
```
Comment (notmandatory) [inline docs/_blog/bindings-scope.md]:
```suggestion
While we started with a simplified version of the Rust BDK API, over time users asked for more and more functionality, and exposing some of the underlying rust-bitcoin constructs became important. This makes sense, and indeed users of the bitcoin development kit in Rust have access to all the related APIs by simply importing rust-bitcoin and rust-miniscript, hence our desire to accommodate these use cases as well. However, this is currently done all in one "bindings" library (i.e. if you import `bdk-android` in a project, you'll have access to an API that is mostly bdk-based, but also contains a bit of rust-bitcoin and rust-miniscript).
```

**Review by notmandatory — APPROVED** [review]
Looks great! I found one little typo but otherwise content should be very useful to other projects who want to get understand what we're doing.

File: docs/_blog/bindings-scope.md (hunk 1 above)
Comment (thunderbiscuit) [reply docs/_blog/bindings-scope.md]: Sweet thanks for finding the typos!

**Comment by DanGould** [comment]
Taking a bit longer to work through than I anticipated. Seems like most systems have python environments, but not Kotlin / Gradle or XCode Developer Tools, so getting started involves some undocumented setup for these examples.

The easiest way to show code examples might be Repl.it

**Comment by thunderbiscuit** [comment]
Good point @DanGould. Do you think it would be better to add documentation to the readmes or to build a tutorial/blog post on how to get set up and ready for building bindings locally? (I could link this tutorial in the readme).

Also, which repos where you taking an interest in (and were hard to set up)? The [uniffi-examples](https://thunderbiscuit.github.io/uniffi-examples/) or the [uniffi-template](https://github.com/thunderbiscuit/uniffi-bindings-template)? They're both in active development and you're sort of my first user haha, so I'd love to hear your thoughts.

**Comment by thunderbiscuit** [comment]
Let's get this in. We can decide what to do with the example repos as they evolve, for now they're under my account and that's fine. I love the Repl.it idea! Don't know enough about it yet but happy to hear more. But this can be added/discussed in other PRs; this one is ready to merge IMO.

**Comment by DanGould** [comment]
we spoke a bit on discord, but I figure I should close the loop here. Before there were links I wasn't looking at your repositories, but since you shred them I've had some success setting up bindings for python and Kotlin. This guide is a useful place to start so I'd love to see it published.

**Pull request merged**